src/platform/published/pal.h

```diff
@@ -301,16 +362,7 @@ constexpr xlang_string_encoding& operator&=(xlang_string_encoding& lhs, xlang_st
     return lhs;
 }
 
-inline constexpr xlang_result xlang_error_ok{ 0 };
-inline constexpr xlang_result xlang_error_mem_invalid_size{ static_cast<int32_t>(0x80080011) };
-inline constexpr xlang_result xlang_error_string_not_null_terminated{ static_cast <int32_t>(0x80000017) };
-inline constexpr xlang_result xlang_error_no_interface{ static_cast<int32_t>(0x80004002) };
-inline constexpr xlang_result xlang_error_pointer{ static_cast<int32_t>(0x80004003) };
-inline constexpr xlang_result xlang_error_sadness{ static_cast<int32_t>(0x80004005) };
-inline constexpr xlang_result xlang_error_out_of_memory{ static_cast<int32_t>(0x8007000e) };
-inline constexpr xlang_result xlang_error_invalid_arg{ static_cast<int32_t>(0x80070057) };
-inline constexpr xlang_result xlang_error_untranslatable_string{ static_cast<int32_t>(0x80070459) };
-inline constexpr xlang_result xlang_error_class_not_available{ static_cast<int32_t>(0x80040111) };
+inline constexpr int32_t xlang_hresult_no_interface{ static_cast<int32_t>(0x80004002) };
 #endif
```

Comment:
I understand that we need E_NOINTERFACE for QI, for interop scenarios.  But it's a bit confusing that we also have xlang_no_interface.  What prevents a QI impl from returning that?  Maybe QI should be strongly typed with a com_interop_result enum that contains two values, including this one?

Reply:
The stance I was thinking of taking is the return value of QueryInterface isn't a xlang error (which would be xlang_error_info*), but rather a convention documented for that function (similar to the return values of AddRef/Release).  If the caller of QueryInterface decides that the value they recieved from it is something they consider an error / unexpected, they can then generate an xlang error for it.  In the case, QueryInterface returns the hresult version of no interface and that is determined by the caller to be an error in their case, they would originate an xlang error with the error as xlang_result::no_interface.

Reply:
I am going to complete this PR, but I do want to continue this conversation to determine whether my stance is the right model and whether it would be clear to a developer implementing QueryInterface or whether it should be changed to something as you mentioned or something else.